src/averageAges.js

```diff
@@ -55,7 +83,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allWomen = people.filter(getGender('f'));
+  const womenWithChildren = allWomen
+    .filter(({ name }) => {
+      return people.some(({ mother }) => {
+        return name === mother;
+      });
+    });
+
+  let childrenToProcess = [];
+
+  !onlyWithSon
+    ? childrenToProcess = people
+      .filter(({ mother }, _, arr) => {
+        return arr.some(({ name }) => {
+          return name === mother;
+        });
+      })
+    : childrenToProcess = people
+      .filter(({ mother, sex }, _, arr) => {
+        return (arr.some(({ name }) => {
+          return name === mother;
+        })) && sex === 'm';
+      });
```

Comment:
let's first find all children and if there is onlyWithSon, we will filter only boys from all the children